app/main.py

```diff
@@ -44,3 +69,45 @@ def main():
 def show_records(db: Session = Depends(get_db)):
     records = db.query(models.EnergyType).all()
     return records
+
+@app.get("/energy-profile/")
+def get_profile_from_postal(postal_code: Optional[str] = None, province: Optional[str] = None, db: Session = Depends(get_db)):
+    records = {}
+    # get postal code and province from energy_profiles depending on input parameters
+    try:
+        if province != None:
+            records['province_name'] = db.query(models.Province).\
+                filter(models.Province.abbreviation == province.upper()).first().name
+            records['energy_profile_code'] = db.query(models.Province).\
+                filter(models.Province.abbreviation == province.upper()).first().energy_profile_code
+        elif postal_code != None and province == None:
+            records['energy_profile_code'] = db.query(models.Province).\
+                filter(models.Province.province_id == postal_to_province[postal_code[0].upper()]).\
+                    first().energy_profile_code
+            records['province_name'] = db.query(models.Province).\
+                filter(models.Province.province_id == postal_to_province[postal_code[0].upper()]).\
+                    first().name
```

Comment:
I think this flow might change in the future to be first look for postal code in the postal codes table (since it would be more granular data, then if it doesn't exist, use the province level data. This is good for now though since we have no postal code level data lol